signature-crate/src/error.rs

```diff
@@ -0,0 +1,25 @@
+//! Signature error types
+
+use core::fmt;
+
+/// Errors which can occur during signature operations
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum Error {
+    /// Signature failed to verify
+    SignatureInvalid,
```

Comment:
Are you sure about having single enum instead of several ZST types? [Here](https://github.com/briansmith/ring/blob/master/src/error.rs#L23) is for example a `ring`'s take on this.

Also it's probably better to implement `std::error::Error` trait behind `std` feature flag.

Reply:
Yes, actually I just wrote up my thoughts about that here:

https://www.reddit.com/r/rust/comments/9wml68/introducing_mundane_a_new_cryptography_library/e9ma3fe/

While what *ring* does is great for *ring*, these traits are intended to also accommodate hardware-backed signers, and therefore need richer error types to express device failures, and are also fallible in ways software implementations are not. 

I can see how a single-member `enum` looks like YAGNI but believe me, I will need it 😉 

There's presently no `std::error::Error` impl, although I can certainly add one upstream in Signatory.

Reply:
Hm, I wonder if it will be sensible to make a signature verification error an associated type. Unfortunately we don't have stable associated type defaults yet, so it will make implementing this trait a bit more verbose.

Reply:
One of my personal goals is trait object safety, which precludes something like that.